melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/KafkaAdminTjeneste.java

```diff
@@ -0,0 +1,126 @@
+import no.nav.melosys.eessi.controller.dto.KafkaConsumerAssignmentResponse;
+import no.nav.melosys.eessi.controller.dto.KafkaConsumerResponse;
+import no.nav.melosys.eessi.identifisering.OppgaveEndretConsumer;
+import no.nav.security.token.support.core.api.Unprotected;
+import org.apache.kafka.common.TopicPartition;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.kafka.config.KafkaListenerEndpointRegistry;
+import org.springframework.kafka.listener.MessageListenerContainer;
+import org.springframework.web.bind.annotation.*;
+
+@Slf4j
+@Unprotected
+@RestController
+@RequestMapping("/admin/kafka/consumers")
+public class KafkaAdminTjeneste {
+
+    private final static String API_KEY_HEADER = "X-MELOSYS-ADMIN-APIKEY";
+
+    @Autowired
+    private KafkaListenerEndpointRegistry kafkaListenerEndpointRegistry;
+
+    private final OppgaveEndretConsumer oppgaveEndretConsumer;
+    private final String apiKey;
+
+    public KafkaAdminTjeneste(OppgaveEndretConsumer oppgaveEndretConsumer, @Value("${melosys.admin.api-key}") String apiKey) {
+        this.oppgaveEndretConsumer = oppgaveEndretConsumer;
+        this.apiKey = apiKey;
+    }
+
+    @GetMapping()
+    public ResponseEntity<List<KafkaConsumerResponse>> hentConsumerIds(@RequestHeader(API_KEY_HEADER) String apiKey) {
+        validerApikey(apiKey);
+
+        return ResponseEntity.ok(kafkaListenerEndpointRegistry.getListenerContainerIds()
+            .stream()
+            .map(this::lagKafkaConsumerResponseVedId)
+            .toList());
+    }
+
+    @PostMapping("/{consumerId}/stop")
+    public ResponseEntity<KafkaConsumerResponse> stoppKafkaConsumer(@PathVariable String consumerId, @RequestHeader(API_KEY_HEADER) String apiKey) {
+        validerApikey(apiKey);
+
+        MessageListenerContainer listenerContainer = kafkaListenerEndpointRegistry.getListenerContainer(consumerId);
+
+        if (listenerContainer == null) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+        }
+
+        listenerContainer.stop();
+
+        return ResponseEntity.ok(lagKafkaConsumerResponse(listenerContainer));
+    }
+
+    @PostMapping("/{consumerId}/start")
+    public ResponseEntity<KafkaConsumerResponse> startKafkaConsumer(@PathVariable String consumerId, @RequestHeader(API_KEY_HEADER) String apiKey) {
+        validerApikey(apiKey);
+
+        MessageListenerContainer listenerContainer = kafkaListenerEndpointRegistry.getListenerContainer(consumerId);
+
+        if (listenerContainer == null) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+        }
+
+        listenerContainer.start();
+
+        return ResponseEntity.ok(lagKafkaConsumerResponse(listenerContainer));
+    }
+
+    @PostMapping("/{consumerId}/seek-to-offset/{offset}")
```

Comment:
`/{consumerId}/seek-to-offset/{offset}` -> `/{consumerId}/seek/{offset}` eventuelt bare `/{consumerId}/{offset}`?

Reply:
/{consumerId}/{offset} it is. Verb i url'en er jo egentlig ikke i henhold:)

Reply:
men det kan lage egne problemer senere, så  /{consumerId}/seek/{offset} it is:)